go/tasks/plugins/k8s/sagemaker/custom_training.go

```diff
@@ -0,0 +1,184 @@
+	inputModeString := strings.Title(strings.ToLower(sagemakerTrainingJob.GetAlgorithmSpecification().GetInputMode().String()))
+
+	jobName := taskCtx.TaskExecutionMetadata().GetTaskExecutionID().GetGeneratedName()
+	outputPath := taskCtx.OutputWriter().GetOutputPrefixPath().String()
+
+	if taskTemplate.GetContainer() == nil {
+		return nil, errors.Errorf("The task template points to a nil container")
+	}
+
+	if taskTemplate.GetContainer().GetImage() == "" {
+		return nil, errors.Errorf("Invalid image of the container")
+	}
+
+	trainingImageStr := taskTemplate.GetContainer().GetImage()
+
+	cfg := config.GetSagemakerConfig()
+
+	var metricDefinitions []commonv1.MetricDefinition
+	idlMetricDefinitions := sagemakerTrainingJob.GetAlgorithmSpecification().GetMetricDefinitions()
+	for _, md := range idlMetricDefinitions {
+		metricDefinitions = append(metricDefinitions,
+			commonv1.MetricDefinition{Name: ToStringPtr(md.Name), Regex: ToStringPtr(md.Regex)})
+	}
+
+	// TODO: When dealing with HPO job, we will need to deal with the following
+	// jobOutputPath := NewJobOutputPaths(ctx, taskCtx.DataStore(), taskCtx.OutputWriter().GetOutputPrefixPath(), jobName)
+
+	hyperParameters, err := injectArgsAndEnvVars(ctx, taskCtx, taskTemplate)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to inject the task template's container env vars to the hyperparameter list")
+	}
+
+	// Injecting a env var to disable statsd for sagemaker tasks
+	statsdDisableEnvVarName := fmt.Sprintf("%s%s%s", FlyteSageMakerEnvVarKeyPrefix, FlyteSageMakerEnvVarKeyStatsdDisabled, FlyteSageMakerKeySuffix)
+	logger.Infof(ctx, "Injecting %v=%v to force disable statsd for SageMaker tasks only", statsdDisableEnvVarName, strconv.FormatBool(true))
+	hyperParameters = append(hyperParameters, &commonv1.KeyValuePair{
+		Name:  statsdDisableEnvVarName,
+		Value: strconv.FormatBool(true),
+	})
+
+	logger.Infof(ctx, "The Sagemaker TrainingJob Task plugin received static hyperparameters [%v]", hyperParameters)
+
+	trainingJob := &trainingjobv1.TrainingJob{
+		Spec: trainingjobv1.TrainingJobSpec{
+			AlgorithmSpecification: &commonv1.AlgorithmSpecification{
+				// If the specify a value for this AlgorithmName parameter, the user can't specify a value for TrainingImage.
+				// in this Flyte plugin, we always use the algorithm name and version the user provides via Flytekit to map to an image
+				// so we intentionally leave this field nil
+				AlgorithmName:     nil,
+				TrainingImage:     ToStringPtr(trainingImageStr),
+				TrainingInputMode: commonv1.TrainingInputMode(inputModeString),
+				MetricDefinitions: metricDefinitions,
+			},
+			// The support of spot training will come in a later version
+			EnableManagedSpotTraining: nil,
+			HyperParameters:           hyperParameters,
+			InputDataConfig:           inputChannels,
+			OutputDataConfig: &commonv1.OutputDataConfig{
+				S3OutputPath: ToStringPtr(outputPath),
+			},
+			CheckpointConfig: nil,
```

Comment:
so this is one thing we might want to model in the future. Passing in a custom checkpoint config?

Reply:
yes we will do this soon